### DISCUSSION TITLE
[Feature Request] Add ability to re-run failed job starting at the failed step

### DISCUSSION THREAD

**Discussion by L1ghtmann** [community/community#38491] [Actions]
Currently, there are two options available following a workflow failure:
![feature request](https://user-images.githubusercontent.com/67040127/200374165-8ca9cc9d-a18c-473e-8ca4-0fb4d27bb70f.png)

Though the first option to re-run failed jobs is often sufficient, there are occasions where a job will fail on a minor step (e.g., a cURL request) later in the job. In such instances, being able to re-run from that failed job's failed step onward would be a huge time saver.   

I did my best to search around the existing discussions and the closest I've found to this request was [here](https://github.com/orgs/community/discussions/27121) where it doesn't seem like a proper solution was found. I apologize in advance if I missed other similar discussions. 

After a quick search, it appears as though someone has made a marketplace action to work around this native limitation (see [here](https://github.com/marketplace/actions/retry-step)), but I'd argue this has enough utility to warrant being a standard actions feature.

**Comment by david-engelmann** [comment]
I feel like it hints at a large issue of intelligent caching for workflow actions
